[Bugfix][setup]: Explicitly include 'mii' package in find_packages

Flink-ddd opened the pull request:
### Description

This PR fixes a packaging issue where `pip install -e .` can fail to discover the top-level `mii` package in certain environments. This leads to a `ModuleNotFoundError: No module named 'mii'` at runtime, making development and testing from a source checkout unreliable.

### Root Cause Analysis

The `setup.py` file relied on `find_packages()` without an explicit `include` argument. While this usually works, it proved to be unreliable in some containerized environments, failing to identify the `mii` directory as an installable package despite the presence of an `__init__.py` file.

### The Fix

To ensure a robust installation, this PR modifies the `packages` argument in `setup.py` to be explicit:

-   **From:** `packages=find_packages(exclude=("tests",))`
-   **To:** `packages=find_packages(include=['mii', 'mii.*'], exclude=("tests",))`

This change explicitly instructs `setuptools` to find and include the `mii` package and all its sub-packages, resolving the installation issue permanently.

### Verification

This packaging fix was verified in a containerized GPU environment (RunPod with the PyTorch 2.1 template).
1.  Cloned the `DeepSpeed-MII` repository.
2.  Applied this change to `setup.py`.
3.  Ran `pip install -e .`.
4.  Successfully executed `python -c "import mii"`, which failed prior to this fix.

### Related Pull Request

This fix is a companion to a bug fix in the core `deepspeed` repository. Resolving this packaging issue was a necessary prerequisite to create a working test environment for the downstream bug.

**The main fix for the inference logic bug is in: [deepspeed/deepspeed#7378](https://github.com/microsoft/deepspeed/pull/7378)**

Flink-ddd committed:
fix(setup): explicitly include mii package to resolve ModuleNotFoundError

Signed-off-by: Vensenmu <vensenmu@gmail.com>

tohtana commented:
Hi @Flink-ddd, 
The change for `find_packages` looks good to me. As the diff of `"` makes it hard to see the intention of this PR, can you revert it to  `'`? Please make sure the formatting is good as well.

Flink-ddd commented:
Hi @tohtana ,

Thanks for the quick review! You got it, I've reverted the quote-style formatting change to make the diff cleaner and only show the substantive change to find_packages.

Just a quick heads-up, the Formatting CI check might fail because of this, as the project's yapf hook seems to automatically enforce double quotes. I've reverted to single quotes as requested for better readability.

Please let me know if there's anything else I can do. Thanks again!

Flink-ddd commented:
Hi @tohtana ,

Thanks a lot for fixing the formatting! I've squashed the formatting commit with my original one and re-applied the DCO sign-off to the combined commit. This should resolve the DCO check failure.

Could you please help approve the workflows when you have a moment? Thank you!

Flink-ddd commented:
Hi @tohtana ,

Thanks for approving the workflows!

I see the nv-a6000-fastgen unit test passed successfully, which is great news, It looks like the other test on the nv-v100-legacy runner was cancelled after timing out in the queue.

Just wanted to give you a heads-up on this CI status. Please let me know if there's anything I can do to help. Thanks again for your time!

tohtana commented:
Hi @Flink-ddd,

As we might need some more time to get the CI runner back, I forcibly merged this PR.
Your change is only about setup.py, and one of the tests that install the package has passed. It shouldn't cause an issue.

Thank you for your contribution!

Flink-ddd commented:
Hi @tohtana ,

That's fantastic news! Thank you so much for looking into the CI issue and for your trust in merging this.

I'm confident this packaging change is safe as it's self-contained, but of course, I'll be on standby to help if any unexpected issues arise.

Thanks again for everything, and happy to contribute in the future!
